api/types.proto

```diff
@@ -713,3 +713,34 @@ message ManagerStatus {
 	// Reachability specifies whether this node is reachable.
 	RaftMemberStatus.Reachability reachability = 4;
 }
+
+// SecretType provides information about what kind of secret this is
+enum SecretType {
+	// CONTAINER secrets are user-specified secrets that should be injected
+	// into the container as an environment variable or file.
+	CONTAINER = 0 [(gogoproto.enumvalue_customname) = "ContainerSecret"];
```

Comment:
I don't agree that we need to flesh this out now. We have a `Type` exactly so we can do different things in the future.

/cc @stevvooe